tests/test_single_topology.py

```diff
@@ -1691,42 +1614,85 @@ def _assert_u_and_grad_consistent(u_fwd, u_rev, x_fwd, x_rev, fused_map):
     )
 
 
-@pytest.mark.nocuda
-@pytest.mark.nightly(reason="slow")
-def test_hif2a_end_state_symmetry_nightly_test():
+def _get_fused_map(mol_a, mol_b, core):
+    amm_fwd = AtomMapMixin(mol_a, mol_b, core)
+    amm_rev = AtomMapMixin(mol_b, mol_a, core[:, ::-1])
+    fused_map = np.concatenate(
+        [
+            np.array([[x, y] for x, y in zip(amm_fwd.a_to_c, amm_rev.b_to_c)], dtype=np.int32).reshape(-1, 2),
+            np.array(
+                [[x, y] for x, y in zip(amm_fwd.b_to_c, amm_rev.a_to_c) if x not in core[:, 0] and y not in core[:, 1]],
+                dtype=np.int32,
+            ).reshape(-1, 2),
+        ]
+    )
+    return fused_map
+
+
+def assert_symmetric_interpolation(mol_a, mol_b, core):
     """
-    Test that end-states are symmetric
+    Assert that the Single Topology interpolation code is symmetric, i.e. ST(mol_a, mol_b, lamb) == ST(mol_b, mol_a, 1-lamb)
+
+    Where for each of the bond, angle, proper torsion, improper torsion, nonbonded, terms
+        - the idxs, params are identical under atom-mapping + canonicalization
+        - u_fwd, u_rev for an arbitrary conformation is identical under atom mapping
+        - grad_fwd, grad_rev for an arbitrary conformation is identical under atom mapping
+
     """
-    with resources.path("timemachine.testsystems.data", "ligands_40.sdf") as path_to_ligand:
-        mols = read_sdf(path_to_ligand)
+    ff = Forcefield.load_default()
+    # map atoms in the combined mol_ab to the atoms in the combined mol_ba
+    fused_map = _get_fused_map(mol_a, mol_b, core)
+
+    st_fwd = SingleTopology(mol_a, mol_b, core, ff)
+    st_rev = SingleTopology(mol_b, mol_a, core[:, ::-1], ff)
+    conf_a = get_romol_conf(mol_a)
+    conf_b = get_romol_conf(mol_b)
+    test_conf = st_fwd.combine_confs(conf_a, conf_b, 0)
+
+    for lamb in np.linspace(0, 1, 12):
```

Comment:
[nit] possibility to strengthen: add special values + randomize? (e.g. `set_prng_seed; lams = [0,0.3,0.5,0.7,1] + list(np.random.rand(0,1,10))` or similar)

(Not sure if there are any informative special-case intermediate lams to included, guessed the stage boundaries from `set(DUMMY_B_CHIRAL_ATOM_CONVERTING_ON_MIN_MAX + DUMMY_B_CHIRAL_ANGLE_CONVERTING_ON_MIN_MAX + ...)` may be useful https://github.com/proteneer/timemachine/blob/ebdb3e4212ae97df4b418c470c726bd3f4f2e6b7/timemachine/fe/single_topology.py#L67-L96 )

Reply:
Good idea.

Reply:
addressed in https://github.com/proteneer/timemachine/pull/1422/commits/5806877e21289caa91bff126db54c501f395af3e